packages/app/src/cli/services/dev/processes/dev-session.ts

```diff
@@ -0,0 +1,209 @@
+  signal: AbortSignal
+}
+
+export interface DevSessionProcess extends BaseProcess<DevSessionOptions> {
+  type: 'dev-session'
+}
+
+export async function setupDevSessionProcess({
+  app,
+  apiKey,
+  developerPlatformClient,
+  ...options
+}: Omit<DevSessionOptions, 'extensions'>): Promise<DevSessionProcess | undefined> {
+  return {
+    type: 'dev-session',
+    prefix: 'extensions',
+    function: pushUpdatesForDevSession,
+    options: {
+      app,
+      apiKey,
+      developerPlatformClient,
+      ...options,
+    },
+  }
+}
+
+export const pushUpdatesForDevSession: DevProcessFunction<DevSessionOptions> = async (
+  {stderr, stdout, abortSignal: signal},
+  options,
+) => {
+  // Force the download of the javy binary in advance to avoid later problems,
+  // as it might be done multiple times in parallel. https://github.com/Shopify/cli/issues/2877
+  const {developerPlatformClient, app} = options
+  await installJavy(app)
+
+  const refreshToken = async () => {
+    return developerPlatformClient.refreshToken()
+  }
+
+  // Create a temporary directory in the system
+  const dir = tempDirectory()
+
+  // Uncomment this to open the temp directory automatically for debugging
+  // await exec(`open`, [dir])
+  const bundlePath = joinPath(dir, 'bundle')
+  await mkdir(bundlePath)
+
+  const processOptions = {...options, stderr, stdout, signal, bundlePath}
+  const appWatcher = new AppEventWatcher(app, processOptions)
+
+  outputWarn('-----> Running DEV on consistentDev mode <-----')
+  outputDebug(`Using temp dir: ${dir}`, stdout)
+  processOptions.stdout.write('Preparing dev session...')
+
+  await initialBuild(processOptions)
+  await bundleExtensionsAndUpload(processOptions, false)
+
+  appWatcher.onEvent(async (event) => {
+    const promises = event.extensionEvents.map(async (eve) => {
+      switch (eve.type) {
+        case EventType.Created:
+        case EventType.UpdatedSourceFile:
+          const message = eve.type === EventType.Created ? '✅ Extension created ' : '🔄 Extension Updated'
+          processOptions.stdout.write(`${message} ->> ${eve.extension.handle}`)
+          return eve.extension.buildForBundle(
+            {...processOptions, app: event.app, environment: 'development'},
+            processOptions.bundlePath,
+            undefined,
+          )
+        case EventType.Deleted:
+          processOptions.stdout.write(`❌ Extension deleted ->> ${eve.extension.handle}`)
+          return rmdir(joinPath(processOptions.bundlePath, eve.extension.handle), {force: true})
+        case EventType.Updated:
+          processOptions.stdout.write(`🔄 Extension Updated ->> ${eve.extension.handle}`)
+          break
+      }
+    })
+    try {
+      await Promise.all(promises)
+      // eslint-disable-next-line no-catch-all/no-catch-all, @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      processOptions.stderr.write('Error building extensions')
+      processOptions.stderr.write(error.message)
```

Comment:
This is temporary, will change when the App Watcher takes the responsibility of doing the building, so i'd ignore it for now. 
And in any case, we don't want to stop `dev` if there is an error building an extension, we should just show the issue so that you can fix it and dev can continue working.